### ISSUE TITLE
Some requests being send to undefined page

### ISSUE THREAD

**Issue by JosephMcArthur** [oaworks/discussion#1582]
Replicatable doing a title search with:

> The efficacy of exercise referral as an intervention for Irish male prisoners presenting with mental health symptoms.

Some citation info for it, if helpful:

International Journal of Prisoner Health. 2018, Vol. 14 Issue 2, p109-123. 15

**Comment by JosephMcArthur** [comment]
I think there may be a dev / live issue here. 

On dev:

`University Faculty’s Awareness and Attitudes towards Open Access Publishing and Institutional Repository: A Case Study` sends me to a request page.

On live, that same search takes me to an undefined one.

**Comment by JosephMcArthur** [comment]
This issue appears to be getting worse. Now getting reports of this with DOIs too.

> Tried searching two different papers with URL, full title, DOI (each a separate search) and same error every time

Those DOIs were:

> 10.1177/1754045313484139
> 10.1111/jsm.12190

I can't debug any further, @markmacgillivray can you take a look at what's happening here please?

**Comment by markmacgillivray** [comment]
This was a pretty weird one, different between dev and live, and appears to be due to differences in underlying data structure. I suspect it is because we have not done any work on requests for quite a while and something somewhere has changed in a small way that we missed. It causes the "fast" creation of a request for the front page of the old website to return an undefined request in some but not all cases. So since we haven't yet prioritised any work on the old requests system, I've fixed this by adding a catch to the frontend UI js for the front page of the old site, where if the response is an undefined request it will do the old fashioned "non-fast" creation of a request, which I've tested and still works fine, it just takes a little longer.

This problem won't exist if we move requests into the new API code, so I don't think it's worth exploring further, as logs show we still get a lot of new requests every day without issue, so it doesn't seem it can be affecting too many.
